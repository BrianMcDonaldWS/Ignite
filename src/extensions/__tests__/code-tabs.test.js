import MarkdownIt from 'markdown-it';
import tabs from '../code-tabs';

const md = new MarkdownIt().use(tabs);

test('code tabs', () => {
  expect(
    md.render(`
::: codeTabs CSS HTML JS

\`\`\`css [0]
.className {
  background: red;
}
\`\`\`

\`\`\`html [1]
<div class="className">
\`\`\`

\`\`\`javascript [2]
function doSomething() {
  return 'foo';
}
\`\`\`

:::
  `)
  ).toMatchSnapshot();
});

test('code tabs - does nothing to non matching blocks', () => {
  expect(
    md.render(`
\`\`\`javascript
function doSomething() {
  return 'foo';
}
\`\`\`
  `)
  ).toMatchSnapshot();
});
