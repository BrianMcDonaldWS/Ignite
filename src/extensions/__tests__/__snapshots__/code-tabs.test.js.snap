// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`code tabs - does nothing to non matching blocks 1`] = `
"<pre><code class=\\"language-javascript\\">function doSomething() {
  return 'foo';
}
</code></pre>
"
`;

exports[`code tabs 1`] = `
"
        <CodeTabs>
          <div class=\\"tabs is-boxed\\">
            <ul>
              
                    <li
                      className=!{this.state.tabIndex === 0 && 'is-active'!}
                      onClick=!{this.onClick(0)!}
                    >
                      <a>
                        <span>CSS</span>
                      </a>
                    </li>
                  

                    <li
                      className=!{this.state.tabIndex === 1 && 'is-active'!}
                      onClick=!{this.onClick(1)!}
                    >
                      <a>
                        <span>HTML</span>
                      </a>
                    </li>
                  

                    <li
                      className=!{this.state.tabIndex === 2 && 'is-active'!}
                      onClick=!{this.onClick(2)!}
                    >
                      <a>
                        <span>JS</span>
                      </a>
                    </li>
                  
            </ul>
          </div>
          <div class=\\"\\">
            <div class=\\"codeBlocks\\" style=!{!{ left: this.state.tabIndex * -100 + '%' !}!}>
      <pre><code data-index=\\"1\\" class=\\"language-css\\">.className {
  background: red;
}
</code></pre>
<pre><code data-index=\\"2\\" class=\\"language-html\\">&lt;div class=&quot;className&quot;&gt;
</code></pre>
<pre><code data-index=\\"3\\" class=\\"language-javascript\\">function doSomething() {
  return 'foo';
}
</code></pre>

          </div>
        </div>
      </CodeTabs>
    "
`;
